### PR TITLE
Write value and detect non-existence correctly

### PR DESCRIPTION
**Pull request by fejta**
/assign @krzyzacy

### CODE REVIEW

File: prow/cmd/gerrit/main_test.go [hunk 1]
```diff
@@ -71,4 +90,32 @@ func TestSyncTime(t *testing.T) {
...
+		opener: fakeOpener{
+			opener:   open,
+			notFound: true,
+		},
+		ctx: ctx,
+	}
+	if err := st.init(); err != nil {
+		t.Fatalf("Failed init: %v", err)
+	}
+	if cur := st.Current(); now.After(cur) {
```
Comment (krzyzacy) [inline prow/cmd/gerrit/main_test.go]: hummmmmm what is this trying to test? I think we should compare against `expected` which is what currently stored in the `path`, but invalidated by `notFound: true`?

Comment (fejta) [reply prow/cmd/gerrit/main_test.go]: This is testing that if GCS returns a 404 we respond the same way as if we do not find a local file (aka fail to time.Now)

Comment (krzyzacy) [reply prow/cmd/gerrit/main_test.go]: I think maybe you want to `st.Update(earlier)` first?

Or the old `now` will always be before `cur` (new now, failure mode in 404) or `later` (whatever stored in `path`)?

Comment (fejta) [reply prow/cmd/gerrit/main_test.go]: Maybe this is more clear

**Pull request merged**